solucao-de-problemas-service/src/main/java/com/ages/incuitech/backend/solucaodeproblemasservice/MailService.java

```diff
@@ -0,0 +1,69 @@
+package com.ages.incuitech.backend.solucaodeproblemasservice;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import javax.mail.*;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.util.List;
+import java.util.Properties;
+
+import static java.util.Objects.isNull;
+
+@Component
+public class MailService {
+    private Session session;
+
+    @Value("${mail.sender.username}")
+    private String username;
+
+    @Value("${mail.sender.password}")
+    private String password;
+
+    @Value("${mail.server.host}")
+    private String serverHost;
+
+    @Value("${mail.server.port}")
+    private String serverPort;
+
+    @Async
+    public void sendTo(String recipient, String subject, String text) {
```

Comment:
"When an @Async method has a Future-typed return value, it is easy to manage an exception that was thrown during the method execution, as this exception is thrown when calling get on the Future result. With a void return type, however, the exception is uncaught and cannot be transmitted. You can provide an AsyncUncaughtExceptionHandler to handle such exceptions." [https://docs.spring.io/spring/docs/current/spring-framework-reference/integration.html#scheduling-annotation-support-exception](url)

Nesse caso eu não vejo a necessidade de criar um handler, mas senti falta de um log de erro no tratamento da exceção. E propagar a unchecked exception nesse caso parece não fazer diferença.

Reply:
Eu alterei ali no catch, como tá em outra linha o git nao pegou a change.